cookbook/controller/argument_value_resolver.rst

```diff
@@ -0,0 +1,234 @@
+.. index::
+   single: Controller; Argument Value Resolvers
+
+Extending Action Argument Resolving
+===================================
+
+.. versionadded:: 3.1
+    The ``ArgumentResolver`` and value resolvers were introduced in Symfony 3.1.
+
+In the book, you've learned that you can get the :class:`Symfony\\Component\\HttpFoundation\\Request`
+object via an argument in your controller. This argument has to be typehinted
+by the ``Request`` class in order to be recognized. This is done via the
+:class:`Symfony\\Component\\HttpKernel\\Controller\\ArgumentResolver`. By
+creating and registering custom argument value resolvers, you can extend
+this functionality.
+
+Functionality Shipped With The HttpKernel
```

Comment:
Shouldn't it be something like this then: `Functionality Shipped with the HttpKernel`? It looks odd to have `with` but `The`